common/djangoapps/course_modes/models.py

```diff
@@ -692,3 +697,38 @@ class CourseModesArchive(models.Model):
     expiration_date = models.DateField(default=None, null=True, blank=True)
 
     expiration_datetime = models.DateTimeField(default=None, null=True, blank=True)
```

Comment:
This isn't true though.  When set it from the signal we want `expiration_datetime_is_explicit` to be false.  Only when it is set by the user via the commerce API do we want to set the `expiration_datetime_is_explicit` boolean to true.